### PR TITLE
Breaking apart complex method

### PR DESCRIPTION
**Pull request by mrprescott**
More sonar-prompted simplification
http://bamboo.s/browse/NX3-OSSF475-1 :white_check_mark:

### CODE REVIEW

**Comment by bradbeck** [comment]
+1

File: plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/proxy/NugetFeedFetcher.java [hunk 1]
```diff
@@ -114,6 +97,35 @@ public Integer cachePackageFeed(final Repository proxy, final URI nugetQuery,
...
+          return new URI(nextPageUrl);
+        }
+        else {
+          log.warn("Page cycle detected: {} -> {}", visited, nextPageUrl);
+        }
+      }
+      return null;
+    }
+    catch (XmlPullParserException e) {
+      throw new IOException("Invalid nuget feed XML received.", e);
```
Comment (jdillon) [inline plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/proxy/NugetFeedFetcher.java]: have to remember log messages are not proper sentences, thus the ending "." is odd, its also not consistent with the log right below it.  Suggest, and have so for a while, that punctuation like this at the end of a log be avoided

Comment (mrprescott) [inline plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/proxy/NugetFeedFetcher.java]: crap, right

**Comment by cstamas** [comment]
+1

**Comment by adreghiciu** [comment]
+1

**Comment by mcculls** [comment]
+1

**Pull request merged**